File: web/src/app/models/response/userAccount.ts
import { AccountStatus } from '../auth';

export interface UserAccount {
  /** The `auth.users` uuid — `public.users.id` is a foreign key onto it. */
  id: string;
  firstName: string;
  lastName: string;
  email: string;
  phone: string;
  companyName: string;
  /** Fingerprint of the PC this licence is claimed by, null until claimed. */
  deviceId: string | null;
  status: AccountStatus;
  createdDate: string;
}

File: src-tauri/src/models/response/user_account.rs
use serde::Serialize;

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct UserAccount {
    pub id: String,
    pub first_name: String,
    pub last_name: String,
    pub email: String,
    pub phone: String,
    pub company_name: String,
    pub device_id: Option<String>,
    pub status: String,
    pub created_date: String,
}
